pkg/util/test/util.go

```diff
@@ -2335,6 +2342,96 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
 	return ValidateTelemetryUninstalled(pxImageList, cluster, timeout, interval)
 }
 
+// GetPortworxVersion returns Portworx version from StorageCluster
+func GetPortworxVersion(cluster *corev1.StorageCluster) (*version.Version, error) {
+	pxVersionTag := strings.Split(cluster.Spec.Image, ":")[1]
+
+	pxVersion, err := version.NewVersion(pxVersionTag)
+	if err != nil {
+		logrus.WithError(err).Warnf("Failed to parse Portworx tag to version, checking version in %s", PxReleaseManifestURLEnvVarName)
+		for _, value := range cluster.Spec.Env {
+			if value.Name == PxReleaseManifestURLEnvVarName {
+				regVersion := regexp.MustCompile(`\/(\d\S+\d)\/version$`)
+				regver := regVersion.FindStringSubmatch(value.Value)
+
+				var ver string
+				if regver != nil {
+					ver = regver[1]
+				}
+
+				if len(ver) == 0 {
+					ver = PxMasterVersion
+					logrus.Warnf("%s=%s doesn't have version set, assuming its latest and setting it to %s", PxReleaseManifestURLEnvVarName, value.Value, ver)
+				} else {
+					logrus.Warnf("Found version in %s=%s, assuming its %s", PxReleaseManifestURLEnvVarName, value.Value, ver)
+				}
+
+				pxVersion, _ = version.NewVersion(ver)
+				return pxVersion, nil
+			}
+		}
+
+		logrus.WithError(err).Warnf("Failed to parse version %s and find %s Env Var in the StorageCluster, assuming its latest and setting it to %s", pxVersionTag, PxReleaseManifestURLEnvVarName, PxMasterVersion)
+		pxVersion, _ = version.NewVersion(PxMasterVersion)
+		return pxVersion, nil
+	}
+
+	logrus.Infof("Portworx version in StorageCluster is %s", pxVersion)
+	return pxVersion, nil
+}
+
+// GetPxOperatorVersion returns PX Operator version
+func GetPxOperatorVersion() (*version.Version, error) {
+	image, err := getPxOperatorImage()
+	if err != nil {
+		return nil, err
+	}
+
+	// We may run the automation on operator installed using private images,
+	// so assume we are testing the latest operator version if failed to parse the tag
+	var versionTag string
+	versionTag = strings.Split(image, ":")[1]
+	if strings.Contains(versionTag, "-dev") {
```

Comment:
with `-dev` we could still parse the version, e.g. version 1.9.0-dev, and can be used to compare versions. so we could skip this step.